File: 2023/python/20231201-trebuchet.py
#!/usr/bin/env python3
from typing import List
import time

file: str = "./2023/data/20231201-trebuchet.txt"

def main():
    sum: int = 0

    with open(file, "r") as f:
        for line in f:
            digits: List[int] = []

            for char in line:
                if char.isdigit():
                    digits.append(char)

            if len(digits) > 0:
                sum += int(digits[0] + digits[-1])

    print(sum)

# Number of iterations
n = 10
total_time = 0

for _ in range(n):
    start_time = time.time()
    main()
    end_time = time.time()
    total_time += (end_time - start_time)

average_time = total_time / n
print("Average execution time over", n, "iterations:", average_time, "seconds")

File: 2023/typescript/20231201-trebuchet.ts
#!/usr/bin/env ts-node
import path from 'path';
import fs from 'fs';

const file: string = path.join(__dirname, '../data/20231201-trebuchet.txt');

function main(): void {
    const lines: string[] = fs.readFileSync(file, 'utf-8').split('\n');
    let sum: number = 0;

    for (const line of lines) {
        if (!line) continue;

        const chars: string[] = line.split('');
        const numbers: string[] = chars.filter((char: string) => {
            return !isNaN(parseInt(char, 10));
        });

        const first: string = numbers[0];
        const last: string = numbers[numbers.length - 1];
        sum += parseInt(first + last);
    }

    console.log(sum);
}

console.time("mainFunction");
main();
console.timeEnd("mainFunction");